app/validators/claim/litigator_claim_validator.rb

```diff
@@ -15,7 +15,35 @@ def self.fields_for_steps
       defendants: [],
       offence_details: %i[offence],
       graduated_fees: %i[actual_trial_length],
+      miscellaneous_fees: %i[defendant_uplifts],
       additional_information: %i[total]
     }
   end
+
+  def validate_defendant_uplifts
+    return if @record.from_api?
+    no_of_defendants = @record.defendants.reject(&:marked_for_destruction?).size
+    add_error(:base, 'lgfs_defendant_uplifts_mismatch') if defendant_uplifts_greater_than?(no_of_defendants)
+  end
+
+  # we add one because uplift quantities reflect the number of "additional" defendants
+  def defendant_uplifts_greater_than?(no_of_defendants)
+    defendant_uplifts_counts.map(&:to_i).any? { |sum| sum + 1 > no_of_defendants }
+  end
+
+  def defendant_uplifts
+    # had to remove basic_fees
+    (@record.fixed_fees + @record.misc_fees).select do |fee|
+      !fee.marked_for_destruction? &&
+        fee.fee_type &&
```

Comment:
**[not a blocker/possible improvement]** 💭 I wonder if we could define a method on the _fee_ itself that answers:

```ruby
fee.fee_type && Fee::BaseFeeType.defendant_uplift_unique_codes.include?(fee.fee_type.unique_code)
```
directly so we avoid the indirection and move the responsibility to check if the fee is _uplifftable_ to the _fee_ itself.

Reply:
👍 this actually already existed (and i had put it in 🤦‍♂️  )